packages/backend-api/src/api/services/updateNotifications.ts

```diff
@@ -58,6 +58,8 @@ const ARTICLE_FIELDS = [...COMMENTSET_FIELDS, 'title', 'url', 'categoryId', 'sou
 
 const ID_FIELDS = new Set(['categoryId', 'tagId', 'ownerId']);
 
+// TODO: Can't find a good way to get rid of the any types below.  And typing is generally a mess.
+//       Revisit when sequelize has been updated
```

Comment:
To convert a big project to be typed, you generally have to do types from the bottom up, with casts to `x as {} as TypeYouWant` to do some explicit type squashing upstream. But it's a very bad code smell if you can't get your types right; it usually means you code is wrong and you don't know it yet, or when you come back later you wont understand you code. So its good to have this TODO here! :)

Reply:
Part of the trouble is that Typescript just can't represent/encode the type munging inherent in functions like pick and the Redux connect function.  Sequelize suffers from this too.  You end up with more typing boilerplate and more unexplained typecasts than code, which IMHO makes your code unmanageable.  

What you really want to be able to do is define your types at runtime, or at least using variables.  E.g., a Pick<> macro that takes an array of strings as its second argument...  Or a function that returns a typed function where the type of the returned function is defined by the arguments to the metafunction.

Reply:
I'm only half convinced; I think you want type-casting at the point where you know what the types are, and when you don't, you put them into named abstractions, typically with a type-parameter that tells you the subtypes. Generally, whenever you are accessing/setting fields, you know something about the structure and I think can and should express that in the input/output types. Having said that, you are also right, there are lots of fancy polymorphic functions that are very hard to type, e.g. linear operators on arbitrary nested arrays can't be typed easily (although you can do recursive things like T = T | Array<T>).